src/script/common/c_content.cpp

```diff
@@ -328,6 +328,11 @@ void read_object_properties(lua_State *L, int index,
 	getfloatfield(L, -1, "zoom_fov", prop->zoom_fov);
 	getboolfield(L, -1, "use_texture_alpha", prop->use_texture_alpha);
 
+	lua_getfield(L, -1, "shaded");
+	if (!lua_isnil(L, -1))
+		prop->shaded = lua_toboolean(L, -1);
```

Comment:
`getboolfield` should work fine here

Reply:
But I need to make it default to true. With getboolfield_default I'll get about the same amount of code

Reply:
What about `getboolfield(L, -1, "shaded", prop->shaded);`?

Reply:
Ok, checked out how getboolfield and it checks for nil internally. Something made me think that it doesn't.

Fixed.